cls_module/cls_module/memory/stm/aha.py

```diff
@@ -185,15 +193,27 @@ def build(self):
     # Build the Pattern Completion (PC) module
     pc_output_shape = self.build_pc(pc_config=self.config['pc'], pc_input_shape=ps_output_shape)
 
-    # Build the Pattern Mapping (PM) module - to the EC (as per biology)
-    self.build_ae(name='pm_ec', config=self.config['pm_ec'],
-                  input_shape=pc_output_shape,
-                  target_shape=self.input_shape)
+    # Build Monosynaptic Pathway
+    if self.config.get('msp_type', None) == 'ca1':
+      self.build_ae(name='ca1', config=self.config['ca1'],
+                    input_shape=self.input_shape,
+                    target_shape=self.input_shape)
+
+      ca1_output_shape = [1, self.config['ca1']['num_units']]
 
-    # Build the Pattern Mapping (PM) module - to an arbitrary 'target' (main use-case is the original input image)
-    self.build_ae(name='pm', config=self.config['pm'],
-                  input_shape=pc_output_shape,
-                  target_shape=self.target_shape)
+      self.build_ae(name='ca3_ca1', config=self.config['ca3_ca1'],
```

Comment:
Since the AE used is a 'simple AE', which is just two FC layers. it is essentially the same thing as a 2 layer FCNN.

The downside is that the AE assumes the number of units on the two layers are equal.

Reply:
That seems like a big downside. Do we have a re-usable component that is a 2 layer FCNN. Any reason not to use it?

Reply:
No, I think at that point we should just construct CA1, etc. as a separate module, but with torch.nn.Linear layers